src/app/core/parsers/zod-entity-parser.service.ts

```diff
@@ -0,0 +1,75 @@
+import { EntityParserService } from '../abstractions/entity-parser.service'
+import { AdLibPiece } from '../models/ad-lib-piece'
+import { BasicRundown } from '../models/basic-rundown'
+import { Part } from '../models/part'
+import { Piece } from '../models/piece'
+import { Rundown } from '../models/rundown'
+import { Segment } from '../models/segment'
+import * as zod from 'zod'
+
+
+export class ZodEntityParser implements EntityParserService {
+    private PIECE_PARSER = zod.object({
+        id: zod.string().nonempty(),
+        partId: zod.string().nonempty(),
+        name: zod.string().nonempty(),
+        layer: zod.string().nonempty(),
+    })
+    public parsePiece(piece: unknown): Piece {
```

Comment:
You are missing new lines before nearly all of your public methods in this file.

Reply:
There are now newlines between all methods in both files.